File: Python Excel Examples/RangesApiDemo/setRangeStyle.py
import spirecloudexcel
from spirecloudexcel.configuration import Configuration as ExcelConfiguration

appId = "your id"
appKey = "your key"
baseUrl="https://api.e-iceblue.cn"
configuration = ExcelConfiguration(appId, appKey,baseUrl)
api = spirecloudexcel.api.ranges_api.RangesApi(configuration)

name = "SetRangeStyle.xlsx"
folder = "Ranges"
storage = ""
sheetName = "Sheet1"
rangeOperate = spirecloudexcel.models.RangeSetStyleRequest()
style = spirecloudexcel.models.Style()
rangeOperate.style = style
style.font = spirecloudexcel.models.Font()
style.font.color = spirecloudexcel.models.Color(255, 255, 0, 0)
style.font.underline = "single"
style.font.size = 12
style.font.name = "楷体"
style.font.is_italic = True
style.font.IsStrikeout = True
style.font.is_bold = True
borderList = list()
border1 = spirecloudexcel.models.Border("Medium", spirecloudexcel.models.Color(255, 255, 0, 0), "EdgeTop")
border2 = spirecloudexcel.models.Border("DashDot", spirecloudexcel.models.Color(255, 0, 255, 0), "EdgeRight")
border3 = spirecloudexcel.models.Border("Thin", spirecloudexcel.models.Color(0, 160, 32, 240), "EdgeLeft")
rangeOperate.style.border_collection = borderList
borderList.append(border1)
borderList.append(border2)
borderList.append(border3)
style.horizontal_alignment = "left"
style.indent_level = 1
style.text_direction = "RightToLeft"
style.background_color = spirecloudexcel.models.Color(0, 255, 255, 0)
style.is_text_wrapped = True
range1 = spirecloudexcel.models.Range()
range1.first_row = 1
range1.first_column = 2
range1.row_count = 10
range1.column_count = 1
rangeOperate.range = range1
api.set_range_style(name, sheetName, rangeOperate, storage=storage, folder=folder)

File: JavaScript Excel Examples/RangesApiDemo/setRangeStyle.js
(function (){
    var appId = "your id";
    var appKey = "your key";
    var baseUrl = "https://api.e-iceblue.cn";

    var SpirecloudExcel = require('../../src/index');
    var apiClient = new SpirecloudExcel.ApiClient(appId, appKey,baseUrl);
    var instance = new SpirecloudExcel.RangesApi(apiClient);
    
    var name = "setRangeStyle.xlsx";
    var opts = {folder: "input" };
    var sheetName = "Sheet1";
    var rangeOperate = new SpirecloudExcel.RangeSetStyleRequest();
    rangeOperate.Style = new SpirecloudExcel.Style();
    rangeOperate.Style.Font = new SpirecloudExcel.Font();
    rangeOperate.Style.Font.Color = new SpirecloudExcel.Color();
    rangeOperate.Style.Font.Color.A = 255;
    rangeOperate.Style.Font.Color.R = 255;
    rangeOperate.Style.Font.Color.G = 0;
    rangeOperate.Style.Font.Color.B = 0;
    rangeOperate.Style.Font.Underline = "single";
    rangeOperate.Style.Font.Size = 12;
    rangeOperate.Style.Font.Name = "Arial";
    rangeOperate.Style.Font.IsItalic = true;
    rangeOperate.Style.Font.IsStrikeout = true;
    rangeOperate.Style.Font.IsBold = true;

    rangeOperate.Style.borderCollection = [new SpirecloudExcel.Border(), new SpirecloudExcel.Border(), new SpirecloudExcel.Border()];
    rangeOperate.Style.borderCollection[0].LineStyle = "Medium";
    rangeOperate.Style.borderCollection[0].Color = new SpirecloudExcel.Color();
    rangeOperate.Style.borderCollection[0].Color.A = 255;
    rangeOperate.Style.borderCollection[0].Color.R = 255;
    rangeOperate.Style.borderCollection[0].Color.G = 0;
    rangeOperate.Style.borderCollection[0].Color.B = 0;
    rangeOperate.Style.borderCollection[0].BorderType = "EdgeTop";

    rangeOperate.Style.borderCollection[1].LineStyle = "DashDot";
    rangeOperate.Style.borderCollection[1].Color = new SpirecloudExcel.Color();
    rangeOperate.Style.borderCollection[1].Color.A = 255;
    rangeOperate.Style.borderCollection[1].Color.R = 0;
    rangeOperate.Style.borderCollection[1].Color.G = 255;
    rangeOperate.Style.borderCollection[1].Color.B = 0;
    rangeOperate.Style.borderCollection[1].BorderType = "EdgeRight";

    rangeOperate.Style.borderCollection[2].LineStyle = "Thin";
    rangeOperate.Style.borderCollection[2].Color = new SpirecloudExcel.Color();
    rangeOperate.Style.borderCollection[2].Color.A = 0;
    rangeOperate.Style.borderCollection[2].Color.R = 160;
    rangeOperate.Style.borderCollection[2].Color.G = 32;
    rangeOperate.Style.borderCollection[2].Color.B = 240;
    rangeOperate.Style.borderCollection[2].BorderType = "EdgeLeft";

    rangeOperate.Style.HorizontalAlignment = "left";
    rangeOperate.Style.indentLevel = 1;
    rangeOperate.Style.textDirection = "RightToLeft";
    rangeOperate.Style.backgroundColor = new SpirecloudExcel.Color();
    rangeOperate.Style.backgroundColor.A = 0;
    rangeOperate.Style.backgroundColor.R = 255;
    rangeOperate.Style.backgroundColor.G = 255;
    rangeOperate.Style.backgroundColor.B = 0;
    rangeOperate.Style.IsTextWrapped = true;

    var range = new SpirecloudExcel.Range();
    range.FirstColumn = 2;
    range.FirstRow = 1;
    range.ColumnCount = 1;
    range.RowCount = 10;
    rangeOperate.Range = range;
    
    instance.setRangeStyle(name, sheetName, rangeOperate, opts,
        function (error, data, response) {
        if (error) throw error;
    });
})();